### PR TITLE
Qnhist

### PR DESCRIPTION
**Pull request by feliciagan**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,4 +31,5 @@ docker-compose down
 - Matching Service: http://localhost:3002
 - Collab Service: http://localhost:3003
 - Code Execution Service: http://localhost:3004
+- Question History Service: http://localhost:3005
```
Comment (jolynloh) [inline README.md:34]: i think nicole is using port 3005 for communication service?

**Review by jolynloh — COMMENTED** [review]
looks good!

File: backend/matching-service/src/utils/api.ts [hunk 2]
```diff
@@ -4,9 +4,20 @@ const QUESTION_SERVICE_URL =
   process.env.QUESTION_SERVICE_URL ||
   "http://question-service:3000/api/questions";
 
+const QN_HISTORY_SERVICE_URL =
+  process.env.QN_HISTORY_SERVICE_URL ||
```
[the remark this answers is not in the document]
Comment (feliciagan) [reply backend/matching-service/src/utils/api.ts:8]: okk included

File: frontend/src/App.tsx [hunk 3]
```diff
@@ -51,6 +52,14 @@ function App() {
                 </ProfileContextProvider>
               }
             />
+            <Route 
+              path="profile/:userId/:qnHistoryId" 
+              element={
+                <AuthProvider>
```
[the remark this answers is not in the document]
Comment (feliciagan) [reply frontend/src/App.tsx]: oh riteee yea dn it here

File: frontend/src/App.tsx [hunk 4]
```diff
@@ -51,6 +52,14 @@ function App() {
                 </ProfileContextProvider>
               }
             />
+            <Route
```
[the remark this answers is not in the document]
Comment (feliciagan) [reply frontend/src/App.tsx]: i tried to nest it but not sure why i cant make it work 😅

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -41,6 +41,7 @@ jobs:
             collab-service,
             code-execution-service,
             communication-service,
+            qn-history-service,
```
Comment (nicolelim02) [inline .github/workflows/ci.yml:44]: i think u need to set up the test environment and files first. bc i got errors running `npm test`

File: backend/qn-history-service/src/models/QnHistory.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
+import mongoose, { Schema, Document } from "mongoose";
+
+export interface IQnHistory extends Document {
+  userIds: string[];
+  questionId: string;
+  title: string;
+  submissionStatus: string;
+  dateAttempted: Date;
+  timeTaken: Number;
```
Comment (nicolelim02) [inline backend/qn-history-service/src/models/QnHistory.ts:9]: would `number` work here? bc theres some errors with `npm run lint`

File: backend/qn-history-service/app.ts
Comment (nicolelim02) [inline backend/qn-history-service/app.ts:1]: think can shift the app.ts, server.ts  and config into src? so the source code are all in the src folder

**Pull request merged**